### ISSUE TITLE
[Meta] Find-in-tabs Feature Requests

### ISSUE THREAD

**Issue by Bamajon1974** [igorlogius/find-in-tabs#2]
Good afternoon. I have had a chance to test Find-in-tabs extension. Overall, the extension works well, is very snappy. 

Below is a list of feature suggestions: 

1. Can you add an option to retain the search results when the search box loses focus? 
2. Can you add the ability for the search to go directly to the first hit once a tab is selected and use up/down arrows to select additional hits on the same page? 
3. The extension needs light and dark mode. 
4. The fonts in the search box and hits might look better if they matched the webpage default font. 
5. The background of the search box might look better if it matched the webpage default background. 
6. In addition to match case, it would be nice to have options to highlight all occurrences, match diacritics, and whole words (same options as default CTRL+F)
7. It would be nice to select custom highlight colors. 
8. For options, you might want to include text upon hovering or underneath or somewhere to identify their function. 

Let me know if you need any clarification for my suggestions above. 

Thank you for your hard work!

**Comment by igorlogius** [comment]
I'll have to slowly work through each one ... but here is some immediate feedback ... which isn't a final verdict but give you an idea what you can expect

1. doable 
2. to make this doable i'll have to change the way how the search is performed ... doable but not something i immediately want to redo after having something that works ... maybe later
3. yes, i might just let the user add some custom CSS, then everyone can make it whatever they want
4. should be solved with custom styling 
5. should be solved with custom styling
6. should be doable ... since i dont yet use the build in search ... not sure how easy this can be done
7. doable
8. [that should already work](https://github.com/igorlogius/find-in-tabs/assets/67047467/4ade6c82-5abd-4a2b-b9b5-ec7d20e0b18f), but having a `(?)` or something might be more obvious as a help indicator

**Comment by Bamajon1974** [comment]
Great. The custom CSS styling is a great idea. At the very least, for interface customization, can you make the extension light/dark theme aware? I never realized how nice/important it is to have dark mode present until I started using Dark Mode throughout Windows...and discovering so many apps and legacy Windows components do not support!

Let me know when you have a new test build to try out!

Thanks!

**Comment by igorlogius** [comment]
when opening the popup and the focus is on the search input area

- pressing up/down arrows changes the number of shown hits)
- hitting Enter in the seach area, switches to the first seach hit and all hits on the site should also be highlighted. (sadly i could not retain the focus on the search input area that is why it seemed better to close it to force the user to reopen it for now. When the input is stored, this should be less of an issue)

next up: retaining the last search input

**Comment by igorlogius** [comment]
- [x] retaining the last search input
- [ ] custom CSS for popup search panel
- [ ] default styling should follow the browser theme (aka. dark/light mode)
- [ ] add "Match Words" function + UI switch 
- [ ] add "Match Diacritics' function + UI switch

**Comment by igorlogius** [comment]
I'll skip the "match word" function, since that doesnt seem to useful at the moment. 

tldr; Basic function are done, the rest is styling ... so i'll close this issue for now.

**Comment by Bamajon1974** [comment]
Great! Thank you! I will test out this evening and let you know.
